NLSY/make_nlsy.R

```diff
@@ -1,24 +1,18 @@
 
-# Execute the script provided by NLSY that reads the data and 
+# Execute the script provided by NLSY that reads the data and
 # save them in an RDS file.
 
 library(tidyverse)
 
-# Get the NLSY data path
-source('common.R')
-
-# The NLSY script has to be executed in the data folder
-setwd(nlsy_data_dir)
-source("College-finance2.R")
-
+# Not sure what packages these functions come from - could not run these three lines
 categories <- vallabels(new_data)
 new_data <- qnames(new_data)
 categories <- qnames(categories)
```

Comment:
Flagging for Damir to library whatever packages are needed for these lines

Reply:
Thank you, Judah, for this work! Those lines require "College-finance2.R", which is in the NLSY data folder. That script is autogenerated by the NLSY website and is contained in the zip file downloaded from there. Every time the data set changes (e.g., when new variables are added), this script needs to be replaced together with data files. This is why I put it there, with the data, to minimize the amount of copying when the data set is revised.

Can we source that script from Box through boxr?

Reply:
Hi Damir, I was able to update `make_nlsy.R` so that everything runs programmatically and pulls in relevant files from Box. It looks like `College-finance2.R` is actually so big and difficult to work with that sourcing it in using `source()` is the only option, but we will need to comment out the first line that loads in the `.dat` file every time it is updated by NLSY (since we now read in the `.dat` file through Box instead).

I saved a new version to Box that reflects that change. Since this script is too big to push to GitHub (truly the biggest R script I have ever seen :) ), I'm fine with continuing to read it in from Box. But if NLSY updates these files frequently, it could get annoying for us.

Reply:
Yes, that script is huge, RStudio can't even open it. Which is one of the reasons it would be good if we can avoid modifying it.

I wonder if we can call boxr::box_setwd() from make_nlsy.R to change working folder to the Box data folder before we source the script. That would allow us to execute the script as is without having to comment anything out.

Reply:
Unfortunately, I looked into this and that function doesn't change your R working directory, just the default directory for `boxr` functions you use. 

One "hacky" idea would be to just write a quick script that downloads the file from Box, deletes the first line, and re-uploads it back using `boxr`. This could be rerun anytime the zip file is updated (as long as the Box links to these files themselves remain unchanged). The `College-finance2.R` is too big to open in RStudio but can still be sourced in from Box without a problem; it will just start running immediately so there has to be an existing file called `new_data` for it to avoid erroring out. Thoughts? 

I also think it's worth saving out the `new_data` to Box so that we only have to go through annoyance this when the data is updated (not sure how often that is)

Reply:
That's disappointing. The function name `box_setwd()` is misleading.

An equally hacky but perhaps more straightforward solution would be to read the script in memory, drop the first 6 lines, i.e., everything before `names(new_data) <- c('E5111701'`, and execute the rest from memory:
```
colfin_link = 'https://urbanorg.box.com/shared/static/wc9eg6bzygf2l7abpjb6sdnd9qyx96tq.r'
colfin_r = readLines(colfin_link)
colfin_r = colfin_r[-1:-6]
tc = textConnection(colfin_r)
source(tc)
close(tc)
```

I am fine with saving `new_data` on Box.

Reply:
Just incorporated this, thanks for suggesting the workaround Damir. I uploaded the data to Box manually - if we need to continue writing out files to Box I can provide the team with our `boxr` credentials and instructions but figured this was easier for now. Otherwise, this PR should be good to merge!

Reply:
That's great, thank you! Merging.